tools/clang/unittests/HLSLExec/ShaderOpArithTable.xml

```diff
@@ -7524,4 +7524,308 @@
             </Parameter>
         </Row>
     </Table>
+    <Table Id="LongVector_BinaryOpTable">
+    <ParameterTypes>
+        <ParameterType Name="ShaderOp.Target">String</ParameterType>
+        <ParameterType Name="ShaderOp.Arguments">String</ParameterType>
+        <ParameterType Name="ShaderOp.Text">String</ParameterType>
+        <ParameterType Name="Validation.Tolerance">double</ParameterType>
+        <ParameterType Array="true" Name="Validation.VecInput1">String</ParameterType>
+        <ParameterType Array="true" Name="Validation.VecInput2">String</ParameterType>
+        <ParameterType Array="true" Name="Validation.ScalarInput">String</ParameterType>
+        <ParameterType Array="true" Name="Validation.VecExpected">String</ParameterType>
+    </ParameterTypes>
+    <Row Name="LongVector_Scalar_Multiply">
+        <Parameter Name="Validation.Tolerance">0</Parameter>
+        <Parameter Name="ShaderOp.Text">
+            <![CDATA[
+
+            template<typename T>
+            struct SLongVectorBinaryOp{
+                T scalarInput;
+                vector<T, NUM> vecInput1;
+                vector<T, NUM> vecInput2;
+                vector<T, NUM> vecOutput;
+            };
+            RWStructuredBuffer<SLongVectorBinaryOp<TYPE> > g_buf : register(u0);
+            [numthreads(8,8,1)]
+            void main(uint GI : SV_GroupIndex) {
+                SLongVectorBinaryOp<TYPE> l = g_buf[GI];
+                l.vecOutput = l.vecInput1 * l.scalarInput;
+                g_buf[GI] = l;
+            };
+
+            ]]>
+        </Parameter>
+        <Parameter Name="ShaderOp.Target">cs_6_9</Parameter>
+        <Parameter Name="Validation.VecInput1">
+            <Value>1,1,1,1</Value>
```

Comment:
Are all these operations elementwise that we're testing?  I wonder if there's a better way to slice it - I'm worried about what'll happen when we come to wanting to test hundreds of elements in a vector.  Also, I'm not sure having every element of the vector is a good test.

I wonder instead if we collect some sample values for each operation and then have the testing infrastructure test different combinations of these with different sized vectors.

Reply:
Yah, I think we will need to get creative here. The test cases have to be reproducible and defining large vectors by hand just isn't going to scale.

My current thoughts:

1. 'Random' generation with the same seed values so it becomes deterministic.
2. 'Fill with sequence' so we can define sequences containing values that we think are interesting, perhaps max/min or whatever in some smaller vector and then fill the test vector with that sequence repeatedly until its filled to the test size.

Reply:
Got rid of using the XML for these tests.